src/services/refactors/inferFunctionReturnType.ts

```diff
@@ -0,0 +1,72 @@
+/* @internal */
+namespace ts.refactor.inferFunctionReturnType {
+    const refactorName = "Infer function return type";
+    const refactorDescription = Diagnostics.Infer_function_return_type.message;
+    registerRefactor(refactorName, { getEditsForAction, getAvailableActions });
+
+    function getEditsForAction(context: RefactorContext): RefactorEditInfo | undefined {
+        const info = getInfo(context);
+        if (info) {
+            const edits = textChanges.ChangeTracker.with(context, t =>
+                t.tryInsertTypeAnnotation(context.file, info.declaration, info.returnType));
+            return { renameFilename: undefined, renameLocation: undefined, edits };
+        }
+        return undefined;
+    }
+
+    function getAvailableActions(context: RefactorContext): readonly ApplicableRefactorInfo[] {
+        const info = getInfo(context);
+        if (info) {
+            return [{
+                name: refactorName,
+                description: refactorDescription,
+                actions: [{
+                    name: refactorName,
+                    description: refactorDescription
+                }]
+            }];
+        }
+        return emptyArray;
+    }
+
+    type ConvertableDeclaration =
+        | FunctionDeclaration
+        | FunctionExpression
+        | ArrowFunction
+        | MethodDeclaration;
+
+    interface Info {
+        declaration: ConvertableDeclaration;
+        returnType: TypeNode;
+    }
+
+    function getInfo(context: RefactorContext): Info | undefined {
+        if (isInJSFile(context.file)) return;
+
+        const token = getTokenAtPosition(context.file, context.startPosition);
+        const declaration = findAncestor(token, isConvertableDeclaration);
+        if (!declaration || declaration.type) return;
+
+        const typeChecker = context.program.getTypeChecker();
+        const type = typeChecker.getTypeAtLocation(declaration);
+        const signatures = type.getCallSignatures();
+        if (!signatures.length) return;
+
+        const returnType = typeChecker.typeToTypeNode(first(signatures).getReturnType(), declaration, NodeBuilderFlags.NoTruncation);
+        if (returnType) {
+            return { declaration, returnType };
+        }
+    }
+
+    function isConvertableDeclaration(node: Node): node is ConvertableDeclaration {

```

Comment:
is it convertable or convertible? help english sucks

Reply:
We have an error "Could not find convertible access expression", and the car is spelt 'convertible', so I'd go with that.

Reply:
@sandersn Thanks, I'll change.